global_helpers/panther_zoom_helpers.py

```diff
@@ -0,0 +1,70 @@
+def get_zoom_user_context(event):
+    """
+    Parses the operation_detail field of Zoom.Operation events related to Users
+    to provide usable fields for use in detections
+    """
+    operation_context = {}
+    raw_string = event.get("operation_detail", "")
+    category_type = event.get("category_type")
+    action = event.get("action")
+
+    if category_type == "User":
+        if action in ("Add", "Delete"):
+            operation_context["User"] = raw_string.split()[2].strip()
+            operation_context["Department"] = raw_string.split("-")[2].split(":")[1].strip()
+            operation_context["UserType"] = raw_string.split("-")[1].split(":")[1].strip()
```

Comment:
are there any cases where the sizes of the resulting array would cause these to raise an exception? Another option is `re.compile()`, but maybe that isn't necessary

Reply:
From what I have seen of the logs the sentence structure of this field is consistent among events that share a category and type so I think we should be OK.

That being said, this particular field in the Zoom logs feels very brittle to me. I had some conversation with Hakmiller about seeing if team ingestion can perhaps do some more extraction from this field upon ingestion. I am going to have that conversation and hopefully update / retire these helpers in the semi-near future.